src-docs/src/components/guide_version_selector/guide_version_selector.tsx

```diff
@@ -0,0 +1,113 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+  SyntheticEvent,
+} from 'react';
+
+import { OuiFlexGroup, OuiFlexItem } from '../../../../src/components/flex';
+import { OuiBadge } from '../../../../src/components/badge';
+import {
+  OuiContextMenuPanel,
+  OuiContextMenuItem,
+} from '../../../../src/components/context_menu';
+import { OuiButton } from '../../../../src/components/button';
+import { OuiPopover } from '../../../../src/components/popover';
+
+export type GuideVersionSelectorProps = {
+  selected: string;
+};
+
+export const GuideVersionSelector: FunctionComponent<GuideVersionSelectorProps> = ({
+  selected,
+}) => {
+  const isLocalDev = window.location.host.includes('803');
+  const [isPopoverOpen, setPopoverOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState<string>(selected);
+  const [options, setOptions] = useState<string[]>([selected]);
+
+  const onButtonClick = () => {
+    setPopoverOpen(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopoverOpen(false);
+  };
+
+  const onChange = useCallback(
+    (value: string) => (e: SyntheticEvent) => {
+      e.preventDefault();
+
+      closePopover();
+      setSelectedOption(value);
+      window.location.href = `/${value}`;
+    },
+    []
+  );
+
+  useEffect(() => {
```

Comment:
nit - does this need to be an effect? The only dep is `isLocalDev`, which we don't expect to change - it's just a static property of the endpoint.

Reply:
Yes! The `useEffect` is important because otherwise we'll be running a `fetch` every time the component is rerendered. Adding `isLocalDev` as a dep is mainly to satisfy the linter, but also a best practice, just in case ;)